routes/index.js

```diff
@@ -457,24 +458,53 @@ function getStnMngmnt(req, res) {
   //set filter based on query parameter passed in
   var data = { authenticated: true }
   var filter = req.params.filter;
-
   filter = (filter === "registered") ? true : (filter === "unregistered") ? false : undefined;
 
-  //get results from database that match filter
-  dbAPI.getStations(filter).then(function (ret) {
-    if (ret !== false && ret !== undefined) {
-      ret.sort(function (a, b) {
-        return (a.registered === b.registered) ? 0 : a.registered ? 1 : -1;
-      });
-      //display sorted array of stations, unregistered stations first
-      data.obj = ret;
-    } else {
-      //display error to user if no database
-      data.obj = ret;
-      req.flash('error', "There was a problem communicating with the database. Please contact the DB administrator.")
-    }
-    res.render('stationManagement.njk', data);
-  });
+  Promise.all([
+    dbAPI.getStations(filter),
+    userStationAPI.getAll()
+  ])
+    .then(
+      ([ret, userStationRel]) => {
+        if (ret !== false && ret !== undefined) {
+          ret.sort(function (a, b) {
+            return (a.registered === b.registered) ? 0 : a.registered ? 1 : -1;
+          });
+          //display sorted array of stations, unregistered stations first
+          data.obj = ret;
+        } else {
+          //display error to user if no database
+          data.obj = ret;
+          req.flash('error', "There was a problem communicating with the database. Please contact the DB administrator.")
+        }
+        var flag = false;
+        var name;
```

Comment:
Could you change this from `var` to `let`? We should be using `let` now instead of `var` since it is safer and has less chance for side effects.